src/mongo/db/s/shard_filtering_metadata_refresh.cpp

```diff
@@ -407,18 +433,22 @@ SharedSemiFuture<void> recoverRefreshShardVersion(ServiceContext* serviceContext
             auto currentMetadata = forceGetCurrentMetadata(opCtx, nss);
 
             if (currentMetadata.isSharded()) {
-                // If migrations are disallowed for the namespace, join any migrations which may be
-                // executing currently
+                // Abort and join any ongoing migration if migrations are disallowed for the
+                // namespace.
                 if (!currentMetadata.allowMigrations()) {
                     boost::optional<SharedSemiFuture<void>> waitForMigrationAbort;
                     {
-                        Lock::DBLock dbLock(opCtx, nss.db(), MODE_IX);
-                        Lock::CollectionLock collLock(opCtx, nss, MODE_IX);
+                        Lock::DBLock dbLock(opCtx, nss.db(), MODE_IS);
+                        Lock::CollectionLock collLock(opCtx, nss, MODE_IS);
```

Comment:
I see you changed the lock mode here 🙂 
I am pretty sure it is fine, but just wanted to check with @smateo that there isn't some reason why we write it with IX here in 6.0.

Reply:
Now that I think about it, this change is not related to SERVER-77309, let me change them back, and we can discuss later on if we can actually change these locks.